pkg/fetch/http.go

```diff
@@ -64,10 +64,16 @@ func (f httpFetcher) File(pathElements ...string) ([]byte, error) {
 }
 
 // FromHTTP returns an object to fetch files from a webserver.
-func FromHTTP(rootURL string, cache bool) (Interface, error) {
-	transport := http.DefaultTransport
+func FromHTTP(transport http.RoundTripper, rootURL string, cache bool) (Interface, error) {
 	if cache {
-		transport = httpcache.NewMemoryCacheTransport()
+		transport = &httpcache.Transport{
+			Transport:           transport,
+			Cache:               httpcache.NewMemoryCache(),
+			MarkCachedResponses: true,
+		}
+	}
+	if transport == nil {
+		transport = http.DefaultTransport
 	}
```

Comment:
Shouldn't this be above the `if cache {`?

Reply:
It shouldn't make a difference, since `httpcache.Transport` will use `http.DefaultTransport` if transport is `nil`. I can see how that may be confusing though, so I moved it before the `if cache {` block.